pkg/db/db.go

```diff
@@ -0,0 +1,51 @@
+package db
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+)
+
+type Resource struct {
+	Kind       string                 `json:"kind"`
+	ApiVersion string                 `json:"apiVersion"`
+	Spec       map[string]interface{} `json:"spec"`
+	Status     map[string]interface{} `json:"status"`
+	Metadata   map[string]interface{} `json:"metadata"`
+}
+
+func (r Resource) Value() (driver.Value, error) {
```

Comment:
It might be worth adding a String() method to Resource as well so that it would also implement the [Stringer interface](https://pkg.go.dev/fmt#Stringer). Since json.Marshall() returns []byte a it would need to do is cast the return from Value() to a string

Reply:
That was part of some tests I've added (the whole `db.go` file. But in this PR I just want to add the things currently needed by the API, so for now I will just add the `Scan` method as I need it to unmarshall the sql records.

But we can keep in mind what you mention